src/WebJobs.Extensions.DurableTask/Bindings/ActivityTriggerAttributeBindingProvider.cs

```diff
@@ -115,39 +113,19 @@ public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
                 object convertedValue;
                 if (destinationType == typeof(object))
                 {
-                    // Straight assignment
                     convertedValue = value;
                 }
-                else
-                {
-                    // Try using the converter manager
-#pragma warning disable CS0618 // Type or member is obsolete
-                    IConverterManager cm = this.parent.extensionContext.Config.ConverterManager;
-#pragma warning restore CS0618 // Type or member is obsolete
-                    MethodInfo getConverterMethod = cm.GetType().GetMethod(nameof(cm.GetConverter));
-                    getConverterMethod = getConverterMethod.MakeGenericMethod(
-                        typeof(DurableActivityContext),
-                        destinationType,
-                        typeof(ActivityTriggerAttribute));
-
-                    Delegate d = (Delegate)getConverterMethod.Invoke(cm, null);
-                    if (d != null)
-                    {
-                        convertedValue = d.DynamicInvoke(value, this.attribute, context);
-                    }
-                    else if (!destinationType.IsInterface)
-                    {
-                        MethodInfo getInputMethod = activityContext.GetType()
-                            .GetMethod(nameof(activityContext.GetInput))
-                            .MakeGenericMethod(destinationType);
-                        convertedValue = getInputMethod.Invoke(activityContext, null);
-                    }
-                    else
-                    {
-                        throw new ArgumentException(
-                            $"Activity triggers cannot be bound to {destinationType}.",
-                            this.parameterInfo.Name);
-                    }
+                else if (destinationType == typeof(DurableActivityContext))
+                {
+                    convertedValue = activityContext;
+                }
+                else if (destinationType == typeof(JObject))
+                {
+                    convertedValue = ActivityContextToJObject(activityContext);
+                }
+                else
+                {
+                    convertedValue = activityContext.GetInput(destinationType);

```

Comment:
Were these changes related to the nuget update? Did some public surface go away?

Reply:
Yes, previously I relied on `IConverterManager`, but that surface area was removed in the latest nuget update and I had to come up with something else. I searched but couldn't find any trigger examples that converted inputs into any arbitrary type. What I ended up realizing though was that I didn't really need the `IConverterManager` stuff at all. Instead I could simply implement my conversions directly in my binding logic. This surprisingly simplified the code quite a bit, though I'm not sure whether there are any downsides to opting-out of using rules/converters for my trigger bindings.

@MikeStall let me know if you have any thoughts on this as well.

Reply:
The new way would be to do something like this: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/Config/CosmosDBConfiguration.cs#L82-L86.

But if this works, let's get it in to unblock.

Reply:
That was the code that I spent the most time looking at, but it only showed how to bind to `string` and `JArray`. My main concern was figuring out how to bind to an unbounded set of arbitrary types.

Reply:
That's what `OpenType` is for. The Cosmos trigger doesn't use it since the types are currently restricted, but the other bindings do -- like here: 
- https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/Config/CosmosDBConfiguration.cs#L65
- https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/Config/CosmosDBConfiguration.cs#L172

But I believe this pattern is to build declarative bindings so tooling can determine the list of supported types. The way you're doing it should still work. If you've tested it, go ahead and get it in. You may want to make a note to come back and refactor to the "new" way in a future update.

Reply:
I've seen that as well, but I've only seen `OpenType` used in the context of collections and not for singular inputs. Anyways, we can follow up on that later. I'll go with what I have for now.